sensors/triggers/params.go

```diff
@@ -203,52 +223,51 @@ func ResolveParamValue(src *v1alpha1.TriggerParameterSource, events map[string]*
 	case src.Value != nil:
 		// Use the default value set by the user in case the event is missing
 		resultValue = *src.Value
-		return &resultValue, nil
+		return &resultValue, stringType, nil
 	default:
 		// The parameter doesn't have a default value and is referencing a dependency that is
 		// missing in the received events. This is not an error and may happen with || conditions.
-		return nil, nil
+		return nil, "", nil
 	}
-
 	// If the event payload parsing failed
 	if err != nil {
 		// Fall back to the default value in case it exists
 		if src.Value != nil {
 			fmt.Printf("failed to parse the event payload, using default value. err: %+v\n", err)
 			resultValue = *src.Value
-			return &resultValue, nil
+			return &resultValue, stringType, nil
 		}
-
 		// Otherwise, return the error
-		return nil, err
+		return nil, "", err
 	}
-
 	// Get the value corresponding to specified key or template within event payload
 	if eventPayload != nil {
 		if tmplt != "" {
 			resultValue, err = getValueWithTemplate(eventPayload, tmplt)
 			if err == nil {
-				return &resultValue, nil
+				return &resultValue, "", nil
 			}
 			fmt.Printf("failed to execute the src event template, falling back to key or value. err: %+v\n", err)
 		}
 		if key != "" {
-			resultValue, err = getValueByKey(eventPayload, key)
+			tmp, typ, err := getValueByKey(eventPayload, key)
+			// For block injection support
+			fmt.Printf("failed to get value by key: %+v\n", err)
```

Comment:
What is this?

How come we have so many `fmt.Printf` in the code.....

Reply:
already been there. 
[https://github.com/argoproj/argo-events/blob/25f5a50df5963b0b5c6c7e0508474ade24b715ba/sensors/triggers/params.go#L240](url)
changing it's place  to after the error check

Reply:
already been there.
[https://github.com/argoproj/argo-events/blob/25f5a50df5963b0b5c6c7e0508474ade24b715ba/sensors/triggers/params.go#L240](https://github.com/argoproj/argo-events/pull/url)
changing it's place to after the error check